### PR TITLE
[Calendar] support more actions after show meeting

### PR DESCRIPTION
**Pull request by KayMKM**
### Purpose
*What is the context of this pull request? Why is it being done?*
close: https://github.com/microsoft/botframework-solutions/issues/2316

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*
1. support all actions after show meetings.
2. add general intent switching in OnInterruptDialogAsync. will switching to other intent when the score over 0.9 and not in show meetings flow.
image

### Tests
*Is this covered by existing tests or new ones? If no, why not?*
added test

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*
support intent switching in other flows

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [ ] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

File: skills/csharp/calendarskill/Dialogs/ShowEventsDialog.cs [hunk 1]
```diff
@@ -612,24 +619,49 @@ private async Task<DialogTurnResult> HandleNextAction(WaterfallStepContext sc, C
                         }
                     }
 
-                    if (state.ShowMeetingInfor.FocusedEvents.Count == 1)
+                    var newFlowOptions = new CalendarSkillDialogOptions() { SubFlowMode = false };
```
Comment (bobokids) [inline skills/csharp/calendarskill/Dialogs/ShowEventsDialog.cs]: move to single function?

Comment (KayMKM) [reply skills/csharp/calendarskill/Dialogs/ShowEventsDialog.cs]: sure. moved.

**Review by bobokids — APPROVED** [review]

**Comment by KayMKM** [comment]
support general intent switching in OnInterruptDialogAsync

File: skills/csharp/calendarskill/Models/CalendarSkillState.cs [hunk 2]
```diff
@@ -40,10 +42,8 @@ public TimeZoneInfo GetUserTimeZone()
         public void Clear()
         {
             UserInfo = new UserInformation();
-            LuisResult = null;
-            GeneralLuisResult = null;
-            PageSize = 0;
             EventSource = EventSource.Other;
+            IsInShowFlow = false;
```
Comment (xhr0804) [inline skills/csharp/calendarskill/Models/CalendarSkillState.cs]: Maybe record something like "InitialIntent" could be more general, and we can reuse it for more intents.

Comment (KayMKM) [reply skills/csharp/calendarskill/Models/CalendarSkillState.cs]: fixed

**Review by xhr0804 — APPROVED** [review]

**Review by lauren-mills — APPROVED** [review]

**Pull request merged**